### PR TITLE
LADX: Full Entrance Rando

### PR DESCRIPTION
**Pull request by zig-for**
## What is this fixing or adding?
Adds full ER to LADX. There's a few places where code isn't as efficient as I'd like, but I don't think it will impact perf on asyncs or anything (though open to someone testing that!)

## How was this tested?
Multiple players have tested

## If this makes graphical changes, please attach screenshots.
No

### CODE REVIEW

**Comment by zig-for** [comment]
Closing for now, but will be back

**Comment by Cybrou** [comment]
We played a dozen of sessions with this merged and tested pretty much all ER options. Everything seems fine 👍.
No logic errors encountered so far.

**Comment by zig-for** [comment]
Bump on this, just played a pretty 17 player LADX only async where half of the players or so had this on, no issues

**Review by ScipioWright — APPROVED** [review]
Test gens succeeded, both with maxed out options and random options. 100 person multiworld test with all random options succeeded, but took a bit longer than I'd like. Unsure if it is because of ER or if it was always like that. I tested with accessibility locations and no locations showed as inaccessible.

I only reviewed init, options, and locations (the main apworld files), and did test gens. I did not review anything in the LADXR folder. I did not test the client.

File: worlds/ladx/__init__.py [hunk 1]
```diff
@@ -1,13 +1,18 @@
 import binascii
 import os
+import copy
+import itertools
+from .Locations import connector_info
+from .LADXR.entranceInfo import ENTRANCE_INFO, entrances_by_type
```
[the remark this answers is not in the document]
Comment (ScipioWright) [reply worlds/ladx/__init__.py:6]: "this" being the `from LADXR.entranceInfo` import. Forgot it shows like 5 lines instead of just the highlighted one.

**Comment by NewSoupVi** [comment]
Zig is no longer actively maintaining LADX.

I'm still willing to merge LADX PRs, but only with a larger volume of peer reviews.

Also, someone needs to sort out the conflicts, obviously

**Comment by ScipioWright** [comment]
Unless Zig wants to continue with this PR, someone else would need to reopen it in their stead (with permission of course)

**Comment by ScipioWright** [comment]
Zig gave me permission to take this PR and make a more merge-ready version. I'll close this after I've put up the new PR

**Pull request closed without merging**